cornell-dti/carriage-web
# Added location table and form

### Summary 






This pull request implements the "add locations" form. As a specific design for where and how admins are to add locations has not been given, the following choices were made:

- A page at `/dashboard/locations` was created with a similar layout to `Riders`.
    - A table displaying current locations (with the option to delete entries).
    - A form to add entries.
- A new type for Locations was defined in `src/types/index.ts`.
- An icon for locations was taken from the `Carriage > Design > Icons-Rider App` google drive folder.

### Test Plan 



1. Head to `/dashboard` and try clicking on the `locations` icon/link in the sidebar.

2. View the location table entries and check with the [backend](https://carriage-web.herokuapp.com/locations).

3. Try adding a location via the form and again check with [backend](https://carriage-web.herokuapp.com/locations).

4. Try deleting the newly added location via the table and again check with [backend](https://carriage-web.herokuapp.com/locations).

![image](https://user-images.githubusercontent.com/52215742/95407324-b2332d80-08ea-11eb-816d-3cf042b7db2f.png)

![image](https://user-images.githubusercontent.com/52215742/95407349-c1b27680-08ea-11eb-9380-f2118204b2fd.png)


### Notes 


- Added a `hidden2` style in `table.css` to avoid changing the already defined `hidden`. This hides the form-feedback without changing the centering of the form.

- Potentially look over the rider form and table code, as there are some implementation improvements.

- There may be a need to implement alerts (for example when deleting an entry fails). In the places of alerts, `TODO` has been placed as a comment... Should comments like these be removed?

### Breaking Changes  


N/A
<!-- - Database schema change (anything that changes DynamoDB document structure)
<!-- - Other change that could cause problems (Detailed in notes)
